### PR TITLE
[Copy] developer guide en edition

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -20,6 +20,65 @@
   - [Learn TiDB SQL](/basic-sql-operations.md)
   - [Learn HTAP](/explore-htap.md)
   - [Import Example Database](/import-example-data.md)
+- Application Develop
+  - Quick Start
+    - [Build a TiDB Cluster in TiDB Cloud(DevTier)](/develop/dev-guide-build-cluster-in-cloud.md)
```
Comment (Oreoxmt) [inline TOC.md:25]:
```suggestion
    - [Build a TiDB Cluster in TiDB Cloud (DevTier)](/develop/dev-guide-build-cluster-in-cloud.md)
```

File: TOC.md [hunk 2]
```diff
@@ -20,6 +20,65 @@
...
+    - Build a Simple CRUD App with TiDB
+      - [Java](/develop/dev-guide-sample-application-java.md)
+  - Developer
+    - [Overview](/develop/dev-guide-overview.md)
+    - Example Applications
+      - [Build the TiDB Application using Spring Boot](/develop/dev-guide-sample-application-spring-boot.md)
+  - Connect to TiDB
+    - [Choose Driver or ORM](/develop/dev-guide-choose-driver-or-orm.md)
+    - [Connect to TiDB](/develop/dev-guide-connect-to-tidb.md)
+    - [Connection Pools and Connection Parameter](/develop/dev-guide-connection-parameters.md)
```
Comment (Oreoxmt) [inline TOC.md:36]:
```suggestion
    - [Connection Pools and Connection Parameters](/develop/dev-guide-connection-parameters.md)
```

File: TOC.md [hunk 3]
```diff
@@ -20,6 +20,65 @@
...
+  - Database Schema Design
+    - [Overview](/develop/dev-guide-schema-design-overview.md)
+    - [Create a Database](/develop/dev-guide-create-database.md)
+    - [Create a Table](/develop/dev-guide-create-table.md)
+    - [Create Secondary Indexes](/develop/dev-guide-create-secondary-indexes.md)
```
Comment (Oreoxmt) [inline TOC.md:41]:
```suggestion
    - [Create a Secondary Index](/develop/dev-guide-create-secondary-indexes.md)
```

File: TOC.md [hunk 4]
```diff
@@ -20,6 +20,65 @@
...
+  - Write Data
+    - [Insert Data](/develop/dev-guide-insert-data.md)
+    - [Update Data](/develop/dev-guide-unstable-result-set.md)
+    - [Delete Data](/develop/dev-guide-delete-data.md)
+    - [Prepared Statement](/develop/dev-guide-prepared-statement.md)
```
Comment (Oreoxmt) [inline TOC.md:46]:
```suggestion
    - [Prepared Statements](/develop/dev-guide-prepared-statement.md)
```

File: TOC.md [hunk 5]
```diff
@@ -20,6 +20,65 @@
...
+  - Read Data
+    - [Get Data from Single Table](/develop/dev-guide-get-data-from-single-table.md)
```
Comment (Oreoxmt) [inline TOC.md:48]:
```suggestion
    - [Query Data from a Single Table](/develop/dev-guide-get-data-from-single-table.md)
```

File: TOC.md [hunk 6]
```diff
@@ -20,6 +20,65 @@
...
+    - [Multi-table Join Queries](/develop/dev-guide-join-tables.md)
+    - [Subqueries](/develop/dev-guide-use-subqueries.md)
```
Comment (Oreoxmt) [inline TOC.md:50]:
```suggestion
    - [Subquery](/develop/dev-guide-use-subqueries.md)
```

File: TOC.md [hunk 7]
```diff
@@ -20,6 +20,65 @@
...
+    - [Connect to TiDB](/develop/dev-guide-connect-to-tidb.md)
+    - [Connection Pools and Connection Parameter](/develop/dev-guide-connection-parameters.md)
+  - Database Schema Design
+    - [Overview](/develop/dev-guide-schema-design-overview.md)
+    - [Create a Database](/develop/dev-guide-create-database.md)
+    - [Create a Table](/develop/dev-guide-create-table.md)
+    - [Create Secondary Indexes](/develop/dev-guide-create-secondary-indexes.md)
+  - Write Data
+    - [Insert Data](/develop/dev-guide-insert-data.md)
+    - [Update Data](/develop/dev-guide-unstable-result-set.md)
```
Comment (Oreoxmt) [inline TOC.md:44]: @Icemap 请确认一下 Update Data 的文件是否为 dev-guide-unstable-result-set.md

File: TOC.md [hunk 8]
```diff
@@ -20,6 +20,65 @@
...
+    - [Common Table Expression](/develop/dev-guide-use-common-table-expression.md)
+    - Read Replica Data
+      - [Follower Read](/develop/dev-guide-use-follower-read.md)
+      - [Stale Read](/develop/dev-guide-use-stale-read.md)
+    - [HTAP Queries](/develop/dev-guide-hybrid-oltp-and-olap-queries.md)
+  - Transaction
+    - [Overview](/develop/dev-guide-transaction-overview.md)
+    - [Optimistic and Pessimistic Transaction](/develop/dev-guide-optimistic-and-pessimistic-transaction.md)
+    - [Transaction Restraints](/develop/dev-guide-transaction-restraints.md)
+    - [Transaction Troubleshoot](/develop/dev-guide-transaction-troubleshoot.md)
```
Comment (Oreoxmt) [inline TOC.md:63]:
```suggestion
    - [Handle Transaction Errors](/develop/dev-guide-transaction-troubleshoot.md)
```

File: TOC.md [hunk 9]
```diff
@@ -20,6 +20,65 @@
...
+  - Troubleshoot
+    - [Overview](/develop/dev-guide-troubleshoot-overview.md)
+    - Others
+      - [Unstable Result Set](/develop/dev-guide-unstable-result-set.md)
+      - [Timeouts](/develop/dev-guide-timeouts-in-tidb.md)
+  - Reference
+    - SQL
+      - [Bookshop Example Application](/develop/dev-guide-bookshop-schema-design.md)
+    - Guidelines
+      - [Object Naming Guidelines](/develop/dev-guide-object-naming-guidelines.md)
```
Comment (Oreoxmt) [inline TOC.md:80]:
```suggestion
      - [Object Naming Convention](/develop/dev-guide-object-naming-guidelines.md)
```

File: TOC.md (hunk 7 above)
Comment (Icemap) [reply TOC.md]: 已更改。

**Comment by Icemap** [comment]
synchronized to https://github.com/pingcap/docs/pull/8735

**Review by Oreoxmt — APPROVED** [review]

File: TOC.md [hunk 10]
```diff
@@ -20,6 +20,65 @@
   - [Learn TiDB SQL](/basic-sql-operations.md)
   - [Learn HTAP](/explore-htap.md)
   - [Import Example Database](/import-example-data.md)
+- Application Develop
```
Comment (qiancai) [inline TOC.md:23]:
```suggestion
- Develop Application
```

File: TOC.md [hunk 11]
```diff
@@ -20,6 +20,65 @@
...
+      - [Java](/develop/dev-guide-sample-application-java.md)
+  - Developer
+    - [Overview](/develop/dev-guide-overview.md)
+    - Example Applications
+      - [Build the TiDB Application using Spring Boot](/develop/dev-guide-sample-application-spring-boot.md)
+  - Connect to TiDB
+    - [Choose Driver or ORM](/develop/dev-guide-choose-driver-or-orm.md)
+    - [Connect to TiDB](/develop/dev-guide-connect-to-tidb.md)
+    - [Connection Pools and Connection Parameters](/develop/dev-guide-connection-parameters.md)
+  - Database Schema Design
```
Comment (qiancai) [inline TOC.md:37]:
```suggestion
  - Design Database Schema 
```

File: TOC.md [hunk 12]
```diff
@@ -20,6 +20,65 @@
...
+  - Transaction
+    - [Overview](/develop/dev-guide-transaction-overview.md)
+    - [Optimistic and Pessimistic Transaction](/develop/dev-guide-optimistic-and-pessimistic-transaction.md)
+    - [Transaction Restraints](/develop/dev-guide-transaction-restraints.md)
+    - [Handle Transaction Errors](/develop/dev-guide-transaction-troubleshoot.md)
+  - Optimize
+    - [Overview](/develop/dev-guide-optimize-sql-overview.md)
+    - [SQL Performance Tuning](/develop/dev-guide-optimize-sql.md)
+    - [Best Practices](/develop/dev-guide-optimize-sql-best-practices.md)
+    - Others
```
Comment (qiancai) [inline TOC.md:68]:
```suggestion
    - Other Optimization Methods
```

File: TOC.md [hunk 13]
```diff
@@ -20,6 +20,64 @@
...
   - [Import Example Database](/import-example-data.md)
+- Develop Application
+  - [Overview](/develop/dev-guide-overview.md)
+  - Quick Start
+    - [Build a TiDB Cluster in TiDB Cloud (DevTier)](/develop/dev-guide-build-cluster-in-cloud.md)
+    - [CRUD SQL in TiDB](/develop/dev-guide-tidb-crud-sql.md)
+    - Build a Simple CRUD App with TiDB
+      - [Java](/develop/dev-guide-sample-application-java.md)
+  - Example Applications
+    - [Build the TiDB Application using Spring Boot](/develop/dev-guide-sample-application-spring-boot.md)
```
Comment (qiancai) [inline TOC.md:31]:
```suggestion
    - [Build a TiDB Application using Spring Boot](/develop/dev-guide-sample-application-spring-boot.md)
```

File: TOC.md [hunk 14]
```diff
@@ -20,6 +20,64 @@
...
+    - [Views](/develop/dev-guide-use-views.md)
+    - [Temporary Tables](/develop/dev-guide-use-temporary-tables.md)
+    - [Common Table Expression](/develop/dev-guide-use-common-table-expression.md)
+    - Read Replica Data
+      - [Follower Read](/develop/dev-guide-use-follower-read.md)
+      - [Stale Read](/develop/dev-guide-use-stale-read.md)
+    - [HTAP Queries](/develop/dev-guide-hybrid-oltp-and-olap-queries.md)
+  - Transaction
+    - [Overview](/develop/dev-guide-transaction-overview.md)
+    - [Optimistic and Pessimistic Transaction](/develop/dev-guide-optimistic-and-pessimistic-transaction.md)
```
Comment (qiancai) [inline TOC.md:60]:
```suggestion
    - [Optimistic and Pessimistic Transactions](/develop/dev-guide-optimistic-and-pessimistic-transaction.md)
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**